### ISSUE TITLE
Feature: Math replacing equality

### ISSUE THREAD

**Issue by MaxOstrowski** [potassco/ngo#76]
In theory, the math trait should be able to similar things than the equality trait.
```
#false :- 1 <= #sum {1,a : a;1,b: b;1,c: c} <= 2, X = #sum {1,e: e;1,f: f;1,g: g} 3, X>=2>1, 5>3.
```
Currently math trait fails here, as it is not able to use 2 comparisons per aggregate, this should be changed and simplified! Maybe the equality trait can be replaced.

**Comment by MaxOstrowski** [comment]
Input:
A list of sympy expressions with 1 relation
Algorithm:
1. Store all relations as lhs relations and the expressions as (semi) linear ast expressions
  - an expression is a factor and something else, and can be constructed from
    - a multiplication with a constant as a factor and something else
    - a constant (factor 1)
    - a variable (factor 1)
    - something else ( factor 1)
2. Find the largest set of agg_ atoms where (at least 2) linear expressions fulfill the following conditions
  - both only have lhs operator
  - both contain exactly the agg_ variables but no other agg_ variables
  - compute the lcm (least common multiple) for the first variable for the factors in both expressions
  - multiply the linear expression with that factor
  - if now all agg_ variables have the same factor in both expressions we can combine both relations into a two sided relation by
  - keeping only the agg_ part in the linear expression
  - moving the rest either to the lhs or rhs
3. combine linear expressions with lhs and rhs to form real literals
Output:
A list of conditions (body) as AST

**Comment by MaxOstrowski** [comment]
A problem that arises is binding variables. Currently, rearranging the linear relation is done using sympy. This can be used to bind variables that need binding (`X = #sum(...)` instread of `0 = #sum(..., -X)`)
But this binding can be destroyed when combining/rearranging relations.

**Comment by MaxOstrowski** [comment]
Simpy function `collect` with `evaluate=False` and `exact=False` can help to figure out common parts regarding aggregate variables. Still
1. rearranging has to be done manually
2. rearranging in general can prevent binding variables

**Comment by MaxOstrowski** [comment]
Rearranging to bind variables is only possible with 1 operator which is Equality.
If we implement it like this, the two rearrangements will never appear in the same relations.

**Comment by MaxOstrowski** [comment]
Closed by 0daad1c349eabce36a85e27c6ab1c37ad11c2f88
